### DISCUSSION TITLE
Can the payload.action passed from a nodered switch be ignored?

### DISCUSSION THREAD

**Discussion by cameronjust** [zachowj/node-red-contrib-home-assistant-websocket#1533] [Q&A] [answered]
I've had some flows that used to work but have recently broken due to a change in either node red or this home assistant plugin.

![Screenshot from 2024-09-08 10-23-12](https://github.com/user-attachments/assets/e4a8dd63-68c3-441e-a997-e523604e431c)

Previously I had data from some buttons going through a switch which then based on the button pressed went to a HA-websocket Action node. The problem now is that the switch passes through the data to HA-Websocket which now breaks as the original data contains payload.action which the HA-Action node throws an error on.

Is there a way to have the HA-Action node ignore the payload.action and just use the configured values for the node?

If there is a better way to achieve this I'm all ears. I did get it working with a Node Red function node but I was trying to avoid Javascript code as it's harder to see what's happening at a glance without looking at the underlying code.

New working method but not ideal

![Screenshot from 2024-09-08 18-13-30](https://github.com/user-attachments/assets/1a8357c5-67b5-4a0f-904c-ee168338f435)

Node Red Function Node code

```
// Extract the action from the incoming MQTT message
let action = msg.payload.action;

// Initialize the output messages (null means no message)
let output1 = null;
let output2 = null;

switch(action) {
    case 'toggle':
        
        // Send to the second output (output2)
        output1 = {
            payload: {
                service: 'toggle'
            }
        };        
        break;
        
    case 'brightness_up_click':
        output2 = {
            payload: {
                service: 'turn_on',
                data: {"brightness_step_pct":10,"transition":2}
            }
        };        
        break;
        
    case 'brightness_down_click':
        output2 = {
            payload: {
                service: 'turn_on',
                data: {"brightness_step_pct":-10,"transition":2}
            }
        };        
        break;
        
    default:
        return null
        break;
}


// Return the message to send it to the next node
return [output1, output2];

```
Ref to my original discussion : https://discourse.nodered.org/t/stop-switch-passing-through-data/90780/4

**Comment by zachowj** [comment, marked as answer]
Currently, the action node does not support ignoring the `msg.payload.action` input override. This feature will likely be added in a future release, similar to the "Block input overrides" option in the current state node.

As a workaround, you can insert a change node between the MQTT node and the switch node. First, move `msg.payload.action` to `msg.action`, then add a second rule to delete `msg.payload.action`. Finally, in the switch node, you can check `msg.action` instead.

**Reply by cameronjust** [reply]
Thanks for the information and possible workaround until "Block input overrides" can be implemented.
